sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -2115,98 +2115,63 @@ def on_add_logical_port(self, port_change_event):
             dict: key is logical port name, value is SFP status
         """
         # A logical port is created. There could be 3 cases:
-        #  1. SFP information is already in DB, which means that a logical port with the same physical index is in DB before.
-        #     Need copy the data from existing logical port and insert it into TRANSCEIVER_DOM_INFO, TRANSCEIVER_STATUS_INFO
-        #     and TRANSCEIVER_INFO table.
-        #  2. SFP information is not in DB and SFP is present with no SFP error. Need query the SFP status by platform API and
+        #  1. SFP is present with no SFP error. Need query the SFP status by platform API and
         #     insert the data to DB.
-        #  3. SFP information is not in DB and SFP is present with SFP error. If the SFP error does not block EEPROM reading,
+        #  2. SFP is present with SFP error. If the SFP error does not block EEPROM reading,
         #     just query transceiver information and DOM sensor information via platform API and update the data to DB; otherwise,
         #     just update TRANSCEIVER_STATUS table with the error.
-        #  4. SFP information is not in DB and SFP is not present. Only update TRANSCEIVER_STATUS_INFO table.
+        #  3. SFP is not present. Only update TRANSCEIVER_STATUS_INFO table.
         logical_port_event_dict = {}
-        sfp_status = None
-        sibling_port = None
         status_tbl = self.xcvr_table_helper.get_status_tbl(port_change_event.asic_id)
         int_tbl = self.xcvr_table_helper.get_intf_tbl(port_change_event.asic_id)
         dom_tbl = self.xcvr_table_helper.get_dom_tbl(port_change_event.asic_id)
         dom_threshold_tbl = self.xcvr_table_helper.get_dom_threshold_tbl(port_change_event.asic_id)
         pm_tbl = self.xcvr_table_helper.get_pm_tbl(port_change_event.asic_id)
-        physical_port_list = self.port_mapping.logical_port_name_to_physical_port_list(port_change_event.port_name)
 
-        # Try to find a logical port with same physical index in DB
-        for physical_port in physical_port_list:
-            logical_port_list = self.port_mapping.get_physical_to_logical(physical_port)
-            if not logical_port_list:
-                continue
+        error_description = 'N/A'
+        status = None
+        read_eeprom = True
+        if port_change_event.port_index in self.sfp_error_dict:
+            value, error_dict = self.sfp_error_dict[port_change_event.port_index]
+            status = value
+            error_bits = int(value)
+            helper_logger.log_info("Got SFP error event {}".format(value))
 
-            for logical_port in logical_port_list:
-                found, sfp_status = status_tbl.get(logical_port)
```

Comment:
The fix looks good to me, and seems to also make the logic cleaner than before. 

Since SfpStateUpdateTask only cares about SW status(the first two fields of STATUS table, which are determined by SW) and DomInfoUpdateTask only cares about HW status (the remaining fields of STATUS table, which are fetched from optical module), just FYI, one easier (but not very clean) fix might be: only consider it as "found", when STATUS table exists as well as its first two fields exist.